avro2py/codegen.py

```diff
@@ -363,10 +362,16 @@ def bridge_namespaces(from_namespace: List[str], to_namespace: List[str], target
         remaining_to_components = [c for c in paired_remaining_to_components if c is not None]
 
         if remaining_from_components:
+            level = len(remaining_from_components)
+            for namespace_name in self.namespaces.keys():
+                namespace_parts = namespace_name.split('.')
+                if namespace_parts[0:len(from_namespace)] == from_namespace and len(namespace_parts) > len(from_namespace):
+                    level += 1
+                    break
```

Comment:
In your PR description, you say:
> then checking if the import-to namespace is a parent of another namespace.
, but I think the way this is written doesn't make it very clear that that's what's occurring here (meaning - I can't very easily follow the logic and see how this code is implementing what you say).

I would probably say, to begin, that we should either have this be a separate method that's named appropriately (like `check_if_to_namespace_is_parent`) or comments added to this snippet describing what we're doing and why.

I would also say, rather than having this be written as an "exceptional" case where we need to check through all the namespaces to see if any qualify as parents, I would much prefer if we pre-computed this upfront for all namespaces, and store this explicitly as some `namespace_is_parent: Dict[Tuple[str], bool]` or `parent_namespaces: FrozenSet[Tuple[str]]` value upon initialization of the class, which we could then just be interrogating here.

Reply:
This logic got changed.